articles/nodejs-timers/index.md

```diff
@@ -0,0 +1,239 @@
+
+**Output**:
+
+```bash
+
+"argument was => John Doe"
+
+```
+
+> NOTE: `setTimeout()` in Node is slightly different from `window.setTimeout()` in JavaScript API as it doesn't accept strings.
+> It's also important to note that `setTimeout()` cannot be fully relied on due to other factors coming into play such as code [blocking](https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/). 
+
+### Scheduling timers using setInterval() method 
+
+Unlike `setTimeout()`, this method is used to execute code multiple times. For instance, [section](section.io) company may send out newsletters to its [Edge as a Service](https://www.section.io/saas-edge-solutions/) customers weekly. This is an example of where the `setInterval()` method comes in. It's an infinite loop that will execute as long as it's not exited (halted).  
+ 
+It has the following syntax: 
+
+```js
+let intervalId = setInterval(callbackFunction, [delay, argument1, argument2, ...]); //option 1
+let intervalId = setInterval(callbackFunction[, delayDuration]); // option 2
+let intervalId = setInterval(code, [delayDuration]); //option 3
+
+```
+
+Let's look at an example:
+
+```js
+function intervalFunction() 
+{ 
+    console.log('This interval is printed after 2 seconds!');
+} 
+
+setInterval(intervalFunction, 2000);
+
+```
+
+Output after 20 seconds: 
+
+```bash
+This interval is printed after 2 seconds!
+This interval is printed after 2 seconds!
+This interval is printed after 2 seconds!
+This interval is printed after 2 seconds!
+This interval is printed after 2 seconds!
+This interval is printed after 2 seconds!
+This interval is printed after 2 seconds!
+This interval is printed after 2 seconds!
+This interval is printed after 2 seconds!
+This interval is printed after 2 seconds! 
+```
+
+In the above example, `intervalFunction()` is being executed every 2 seconds until it's exited (stopped).
+
+### Scheduling timers using setImmediate() method 
+
+The `setImmediate()` method is used to execute code at the end of the loop cycle.
+In simple terms, this method breaks tasks that take longer to execute to run a callback function initiated by other operations such as events.
+
+`setImmediate()` function has the following syntax: 
+
+``` js
+let immediateId = setImmediate(callbackFunction, [param1, param2, ...]);
+let immediateId = setImmediate(callbackFunction);
+```
+
+Let's look at an example: 
+
+```js
+console.log('before set immediate function is called'); 
+
+setImmediate((arg) => 
+{ 
+    console.log(`executing the immediate function: ${arg}`);
+}); 
+
+console.log('after immediate function has been executed');
+``` 
+
+**Output**:
+
+```bash
+before a set immediate function is called
+after the immediate function has been executed
+executing the immediate function: undefined
+```
+
+While executing this method, you're likely to encounter an error as seen below, otherwise, skip this part:
```

Comment:
Please do explain why this error is likely to be encountered in a few lines.
And please also explain why the fix you suggested works that way. Readers would get a deeper understanding of why `setImmediate()` breaks and throws an error in some cases.